File: service/src/main/java/com/caldremch/andorid/coponent/service/IServiceCallback.java
package com.caldremch.andorid.coponent.service;

import androidx.annotation.NonNull;

/**
 * @author Caldremch
 * @date 2019-06-03 18:27
 * @email caldremch@163.com
 * @describe
 **/
public interface IServiceCallback<T> {
    void onService(@NonNull T service);
    default void noService(){}
}

File: service-core/src/main/java/com/caldremch/andorid/coponent/service/IServiceCallback.kt
package com.caldremch.andorid.coponent.service

/**
 * @author Caldremch
 * @date 2019-06-03 18:27
 * @email caldremch@163.com
 * @describe
 */
interface IServiceCallback<T> {
    fun onService(service: T)
    fun noService() {}
}